src/convert.c

```diff
@@ -221,7 +221,7 @@ int convertToLeader(struct raft *r)
         r->barrier.type = RAFT_BARRIER;
         r->barrier.term = r->current_term;
         r->barrier.buf.len = 8;
-        r->barrier.buf.base = raft_malloc(r->barrier.buf.len);
+        r->barrier.buf.base = raft_calloc(1, r->barrier.buf.len);
```

Comment:
This memory is effectively just a placeholder, and it does not matter if it's initialized or not .That's probably what valgrind complained about.

It's fine to use `calloc` here to silence the warning. Would you mind also adding a comment saying that calloc is being used for this reason?